pkg/traceql/ast_execute_test.go

```diff
@@ -676,3 +703,29 @@ func BenchmarkBinOp(b *testing.B) {
 		})
 	}
 }
+
+// BenchmarkUniquespans benchmarks the performance of the uniqueSpans function using
+// different numbers of spansets and spans.
+func BenchmarkUniqueSpans(b *testing.B) {
```

Comment:
Neat, did the benchmarks show a benefit from the current approach?  It seems like it would be the same number of insertions and lookups at first glance.

Reply:
I tried building the dictionary using the first, largest and smallest spanset. Smallest outperformed the others. Here is first -> smallest:

```
name                       old time/op    new time/op    delta
UniqueSpans/10000|1-8         152µs ± 1%     169µs ±15%  +11.23%  (p=0.032 n=5+5)
UniqueSpans/100|1-8          2.20µs ±23%    2.25µs ± 5%     ~     (p=0.968 n=5+5)
UniqueSpans/1|1-8             112ns ±12%     112ns ±18%     ~     (p=0.738 n=5+5)
UniqueSpans/10000|100-8       364µs ±15%     345µs ±10%     ~     (p=0.548 n=5+5)
UniqueSpans/100|100-8        12.5µs ±21%    14.5µs ±17%     ~     (p=0.095 n=5+5)
UniqueSpans/1|100-8          9.80µs ±11%    2.01µs ± 7%  -79.47%  (p=0.008 n=5+5)
UniqueSpans/10000|10000-8    1.62ms ±12%    1.60ms ±13%     ~     (p=1.000 n=5+5)
UniqueSpans/100|10000-8      1.45ms ±13%    0.39ms ± 6%  -73.18%  (p=0.008 n=5+5)
UniqueSpans/1|10000-8        1.21ms ± 7%    0.23ms ±25%  -81.21%  (p=0.008 n=5+5)

name                       old alloc/op   new alloc/op   delta
UniqueSpans/10000|1-8         164kB ± 0%     164kB ± 0%     ~     (all equal)
UniqueSpans/100|1-8          1.79kB ± 0%    1.79kB ± 0%     ~     (all equal)
UniqueSpans/1|1-8             32.0B ± 0%     32.0B ± 0%     ~     (all equal)
UniqueSpans/10000|100-8       169kB ± 0%     169kB ± 0%     ~     (p=0.992 n=5+5)
UniqueSpans/100|100-8        8.43kB ± 0%    8.42kB ± 0%   -0.02%  (p=0.000 n=4+5)
UniqueSpans/1|100-8          7.02kB ± 0%    1.79kB ± 0%     ~     (p=0.079 n=4+5)
UniqueSpans/10000|10000-8    1.00MB ± 0%    1.00MB ± 0%     ~     (p=0.548 n=5+5)
UniqueSpans/100|10000-8       840kB ± 0%     169kB ± 0%  -79.87%  (p=0.008 n=5+5)
UniqueSpans/1|10000-8         840kB ± 0%     164kB ± 0%  -80.49%  (p=0.008 n=5+5)

name                       old allocs/op  new allocs/op  delta
UniqueSpans/10000|1-8          1.00 ± 0%      1.00 ± 0%     ~     (all equal)
UniqueSpans/100|1-8            1.00 ± 0%      1.00 ± 0%     ~     (all equal)
UniqueSpans/1|1-8              1.00 ± 0%      1.00 ± 0%     ~     (all equal)
UniqueSpans/10000|100-8        9.00 ± 0%      9.00 ± 0%     ~     (all equal)
UniqueSpans/100|100-8          9.00 ± 0%      9.00 ± 0%     ~     (all equal)
UniqueSpans/1|100-8            9.00 ± 0%      1.00 ± 0%  -88.89%  (p=0.008 n=5+5)
UniqueSpans/10000|10000-8       214 ± 0%       214 ± 0%     ~     (all equal)
UniqueSpans/100|10000-8         214 ± 0%         9 ± 0%     ~     (p=0.079 n=4+5)
UniqueSpans/1|10000-8           214 ± 0%         1 ± 0%  -99.53%  (p=0.008 n=5+5)
```